kahuna/public/js/search/query.js

```diff
@@ -62,6 +69,16 @@ query.controller('SearchQueryCtrl',
     Object.keys($stateParams)
           .forEach(setAndWatchParam);
 
+    ctrl.changeQuery = () =>
+        searchQueryService.set(ctrl.filter.query);
+
+    // FIXME: This is here to stop circular injection of the model.
+    // Avoiding: queryChange => set() => emit() => queryChange()
+    searchQueryService.q$.filter(q =>
+        q !== ctrl.filter.query
```

Comment:
Is this necessary? I would have thought that if `q === ctrl.filter.query`, then `ctrl.filter.query = q` is essentially a no-op?